libcudacxx/include/cuda/std/detail/libcxx/include/__iterator/move_sentinel.h

```diff
@@ -0,0 +1,65 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// SPDX-FileCopyrightText: Copyright (c) 2023 NVIDIA CORPORATION & AFFILIATES
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _LIBCUDACXX___ITERATOR_MOVE_SENTINEL_H
+#define _LIBCUDACXX___ITERATOR_MOVE_SENTINEL_H
+
+#ifndef __cuda_std__
+#include <__config>
+#endif // __cuda_std__
+
+#include "../__concepts/assignable.h"
+#include "../__concepts/convertible_to.h"
+#include "../__concepts/semiregular.h"
+#include "../__utility/move.h"
+
+#if defined(_LIBCUDACXX_USE_PRAGMA_GCC_SYSTEM_HEADER)
+#pragma GCC system_header
+#endif
+
+#if _LIBCUDACXX_STD_VER > 14
+
+_LIBCUDACXX_BEGIN_NAMESPACE_STD
+
+#if _LIBCUDACXX_STD_VER > 17
+template <semiregular _Sent>
+#else
+template<class _Sent, enable_if_t<semiregular<_Sent>, int> = 0>
+#endif
+class _LIBCUDACXX_TEMPLATE_VIS move_sentinel
+{
+public:
+  _LIBCUDACXX_HIDE_FROM_ABI
+  move_sentinel() = default;
```

Comment:
question: should it be marked `constexpr` as well?

Reply:
It should not be necessary, unless you test against old msvc and ctk 11.1

Added constexpr

Reply:
We discussed this internally and decided to add the constexpr